docs/e2e_tutorial.rst

```diff
@@ -0,0 +1,280 @@
+++++++++++++++++++++
+
+As mentioned before, we can introduce a fixture which will enable us to poll until harvest was
+finished, instead of having a simple ``time.sleep`` (snippet taken from
+``tests/e2e/test_arxiv_harvest.py``):
+
+.. code-block:: python
+
+    def wait_for(func, *args, **kwargs):
+        max_time = kwargs.pop('max_time', 200)
+        interval = kwargs.pop('interval', 2)
+
+        decorator = backoff.on_exception(
+            backoff.constant,
+            AssertionError,
+            interval=interval,
+            max_time=max_time,
+        )
+        decorated = decorator(func)
+        return decorated(*args, **kwargs)
+
+We can then use the fixture in our test:
+
+.. code-block:: python
+
+    @with_mitmproxy
+    def test_arxiv_in_hp(inspire_client, mitm_client):
+        inspire_client.e2e.schedule_crawl(
+            spider='arXiv_single',
+            workflow='article',
+            url='http://export.arxiv.org/oai2',
+            identifier='oai:arXiv.org:1806.04664',
+        )
+
+        def _in_holdinpen():
+            holdingpen_entries = inspire_client.holdingpen.get_list_entries()
+            assert len(holdingpen_entries) > 0
+            assert holdingpen_entries[0].status == 'HALTED'
+            return holdingpen_entries
+
+        # Will poll every two seconds and timeout after 200 seconds
+        holdingpen_entries = wait_for(_in_holdinpen)
+
+        assert len(holdingpen_entries) == 1
+
+        holdingpen_entry = holdingpen_entries[0]
+
+        assert holdingpen_entry.core is None
+        assert holdingpen_entry.arxiv_eprint == '1806.04664'
+
+We can also use the mitmproxy client to make assertions on the interactions with external services
+that happened during our test:
+
+.. code-block:: python
+
+    @with_mitmproxy
+    def test_arxiv_in_hp(inspire_client, mitm_client):
+        # ... ...
+        mitm_client.assert_interaction_used('ArxivService', 'interaction_0', times=1)
+
+Above will fail if the interaction ``scenarios/arxiv_in_hp/ArxivService/interaction_0.yaml`` has not
+been used exactly one time. You can leave off the ``times`` parameter if you want to assert that
+the interaction happened at least once, instead of specifying exactly the number of times. Names
+of interactions are not important so you can rename them if you like. Naming only matters if two
+interactions can match the same request: in such case the lexicographically first one is chosen for
+consistency.
+
+Troubleshooting/Tips
+++++++++++++++++++++
+
+Accessing web node in browser
+-----------------------------
+
+If for any reason you need to access the web interface of INSPIRE, you can add an entry to your
+``/etc/hosts`` file with the IP of the web container:
+
+.. code-block:: bash
+
+    $ docker inspect inspirenext_test-web-e2e.local_1 | grep '"IPAddress"'
+
+                "IPAddress": "",
+                    "IPAddress": "172.20.0.9",
+
+    $ sudo vim /etc/hosts
+
+And add a line add the bottom:
```

Comment:
@szymonlopaciuk I know you're bit of a perfectionist, so to bug you I'm pointing out that there's a typo here.

Reply:
Thank you, @Dinika :smile: